lib/matplotlib/backend_bases.py

```diff
@@ -3264,15 +3266,16 @@ def draw_if_interactive(cls):
 
     @classmethod
     @cbook._make_keyword_only("3.1", "block")
-    def show(cls, block=None):
+    def show(cls, block=None, *, figures=None):
         """
         Show all figures.
 
         `show` blocks by calling `mainloop` if *block* is ``True``, or if it
         is ``None`` and we are neither in IPython's ``%pylab`` mode, nor in
         `interactive` mode.
         """
-        managers = Gcf.get_all_fig_managers()
+        managers = ([figure.canvas.manager for figure in figures]
```

Comment:
This should be more forgiving to canvases without managers?

Reply:
I don't think ignoring (silently or not) figures we can't show because they have no manager is really helpful.